haystack/agents/conversational.py

```diff
@@ -1,63 +1,87 @@
 from typing import Optional, Callable, Union
+import warnings
 
+from haystack.agents.base import ToolsManager, react_parameter_resolver
 from haystack.agents import Agent
 from haystack.agents.memory import Memory, ConversationMemory
 from haystack.nodes import PromptNode, PromptTemplate
 
 
 class ConversationalAgent(Agent):
     """
-    A conversational agent that can be used to build a conversational chat applications.
+    A ConversationalAgent is an extension of the Agent class that supports the use of tools in
+    conversational chat applications. This agent can make use of the ToolsManager to manage a set of tools and seamlessly integrate them into the conversation.
 
     Here is an example of how you can create a simple chat application:
     ```
     import os
 
-    from haystack.agents.base import ConversationalAgent
-    from haystack.agents.memory import ConversationSummaryMemory
+    from haystack.agents.conversational import ConversationalAgent
     from haystack.nodes import PromptNode
+    from haystack.tools import ToolsManager, Tool
 
-    pn = PromptNode("gpt-3.5-turbo", api_key=os.environ.get("OPENAI_API_KEY"), max_length=256)
-    agent = ConversationalAgent(pn, memory=ConversationSummaryMemory(pn))
+    # Initialize a PromptNode and a ToolsManager with the desired tools
+    prompt_node = PromptNode("gpt-3.5-turbo", api_key=os.environ.get("OPENAI_API_KEY"), max_length=256)
+    tools_manager = ToolsManager([Tool(name="ExampleTool", pipeline_or_node=example_tool_node)])
 
+    # Create the ConversationalAgent instance
+    agent = ConversationalAgent(prompt_node, tools_manager=tools_manager)
+
+    # Use the agent in a chat application
     while True:
         user_input = input("Human (type 'exit' or 'quit' to quit): ")
         if user_input.lower() == "exit" or user_input.lower() == "quit":
             break
-        elif user_input.lower() == "memory":
-            print("\nMemory:\n", agent.memory.load())
         else:
             assistant_response = agent.run(user_input)
             print("\nAssistant:", assistant_response)
-
     ```
     """
 
     def __init__(
         self,
         prompt_node: PromptNode,
         prompt_template: Optional[Union[str, PromptTemplate]] = None,
-        memory: Optional[Memory] = None,
-        prompt_parameters_resolver: Optional[Callable] = None,
+        tools_manager: Optional[ToolsManager] = None,
+        memory: Optional[Memory] = ConversationMemory(),
+        prompt_parameters_resolver: Optional[Callable] = react_parameter_resolver,
```

Comment:
To support users who want to run ConversationalAgent without any tool, we could log a warning message in the Agent run method. If the Agent's run method is called while there are no tools and the prompt_parameters_resolver is set to react_parameter_resolver and the final answer pattern is set to r"Final Answer\s*:\s*(.*)", we could suggest other settings in the warning message.